libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -238,64 +265,122 @@ string IRGenerator::generateInitialAssignment(VariableDeclaration const& _varDec
 	return generator.code();
 }
 
-string IRGenerator::constructorCode(ContractDefinition const& _contract)
+pair<string, map<ContractDefinition const*, string>> IRGenerator::evaluateConstructorArguments(
+	ContractDefinition const& _contract
+)
 {
-	// Initialization of state variables in base-to-derived order.
-	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
-
-	using boost::adaptors::reverse;
-
-	ostringstream out;
-
-	FunctionDefinition const* constructor = _contract.constructor();
-	if (constructor && !constructor->isPayable())
-		out << callValueCheck();
+	map<ContractDefinition const*, string> constructorParams;
+	vector<pair<ContractDefinition const*, std::vector<ASTPointer<Expression>>>> baseConstructorArguments;
+	if (FunctionDefinition const* constructor = _contract.constructor())
+		for (auto const& modifier: constructor->modifiers())
+		{
+			FunctionDefinition const* baseConstructor = dynamic_cast<ContractDefinition const*>(
+				modifier->name()->annotation().referencedDeclaration
+			)->constructor();
+
+			if (baseConstructor && modifier->arguments() && !modifier->arguments()->empty())
+				baseConstructorArguments.emplace_back(
+					dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
+					*modifier->arguments()
+				);
+		}
```

Comment:
The part in line 281 is still duplicated. What about
```suggestion
			if (FunctionDefinition const* baseConstructor = dynamic_cast<ContractDefinition const*>(
				modifier->name()->annotation().referencedDeclaration
			)->constructor())
				baseConstructorArguments.emplace_back(
					dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
					modifier->arguments()
				);
		}
```